pyriemann/utils/utils.py

```diff
@@ -134,3 +142,34 @@ def check_metric(metric, expected_keys=["mean", "distance"]):
 
     else:
         raise TypeError("metric must be str or dict, but got {type(metric)}")
+
+
+def check_function(fun, functions):
+    """Check the function to use.
+
+    Parameters
+    ----------
+    fun : string | callable
+        Function to check.
+    functions : dict
+        Functions available in API.
+
+    Returns
+    -------
+    fun : callable
+        Function to use.
+
+    Notes
+    -----
+    .. versionadded:: 0.6
+    """
+    if isinstance(fun, str):
+        if fun not in functions.keys():
+            raise ValueError(f"Unknown function name '{fun}'. Must be one of "
+                             f"{' '.join(functions.keys())}")
+        else:
+            fun = functions[fun]
+    elif not hasattr(fun, '__call__'):
+        raise ValueError("Argument must be a string or a callable "
+                         f"(Got {type(fun)}).")
+    return fun
```

Comment:
The case "callable" allows users to apply their own function, which may not belong to pyRiemann.

Reply:
Ok. In this case, I suggest to modify a little bit the doc. Something along these lines:


fun: ... If the function is a callable, it can be a function defined in functions or a function defined by the user. In this case, the signature of the function as to match the ones defined in functions. This is the user responsibility to ensure this, and will not be checked.


functions:... This parameter is only used when fun is a string.

Reply:
You are right! I update the doc.